### ISSUE TITLE
synchronize between tabs

### ISSUE THREAD

**Issue by mezod** [rt2zz/redux-persist#485]
Hi,

I have read through the docs and wanted to ask a couple questions to wrap my head around it.

I understand that autoRehydrate is just to repopulate the redux state on store creation (normally, for simple apps, when app is launched).

Is there a way to autoRehydrate on localStorage change by a third party? Let me explain.
I have an application that stores reduxState into localStorage, so that when a new tab is open, the data can be immediately used to populate the new tab/instance of the app. Then, if through interaction, reduxState of the new tab app is changed, I update localStorage. I wonder, how could I update the app instance from the initial tab? 

I know from the spec (https://html.spec.whatwg.org/multipage/webstorage.html#the-localstorage-attribute) that when localStorage is changed in one tab it issues an event to the other tabs. So my final question would be, is this something redux-persist considers at all? And assuming not, is there a reasonable work around using redux-persist?

Thank you!

**Comment by rt2zz** [comment]
You might look at / try https://github.com/rt2zz/redux-persist-crosstab

It uses the exact event to sync between tabs.

**Comment by mezod** [comment]
thank you!

Maybe it would be a good idea to add a link to it on the docs for this repo! :)

**Comment by rt2zz** [comment]
right, I have been hesitant only because I have not used it in prod / do not know what limitations exist. *but* I know others have used it in prod successfully.

Next step is to figure out how it would work with the forthcoming redux-persist@5

**Comment by mezod** [comment]
I can't help you with that right now, still struggling to fully understand how to get redux-persist to work :P but if it does, there'll be another instance of redux-persist-crosstab in production :)

**Comment by mezod** [comment]
Hey, just wanted to let you know that both libs work wonderfully!

I googled what "out of wack" means and is it possible that you meant "out of whack" (in the purge section)? (I didn't know this phrase in any case). What do you mean with that anyways? I hope it doesn't go out of whack!!

I understand purge is to remove all the keys from localStorage. I need to do so when the user logs out. Since log out takes place in an action creator, which do you think would be the best way to have access to the store from there to be able to persistStore(store).purge()? I used to use getState() but it seems persistStore requires the store. Importing the singleton doesn't sound like a good idea either, right?

**Comment by rt2zz** [comment]
persistor as a singleton is one option. In general singletons are discouraged but the persistor typically only makes sense as 1 persistor per process, and is not typically used during SSR, so in this case singleton might just work.  Your other option is I think to send persistor along with whatever calls your action - but that requires more work to expose persistor via context.

**Comment by mezod** [comment]
thank you @rt2zz! feel free to close this :-)

**Comment by mezod** [comment]
hey @rt2zz one last doubt, I promise! When I use persistor.purge() it successfully removes all the keys from the localStorage except 

`Storage {reduxPersist:basedate: "null", length: 1}`

basedate is just a stringified date. When I purge it, it becomes null, but the key isn't removed from the localStorage. Why is it so?

**Comment by rt2zz** [comment]
yes purge should be removing basedate. Perhaps basedate is getting changed to null immedaitely after purge which cases persist to occur for this key? Do you have any code that sets basedate to null explicitly and if so can you try and trace out if it is being called *after* purge?

**Comment by mezod** [comment]
Your reasoning made a lot of sense, but I never set this key to null in my app.

It's either initialized with:

`const initialState = moment().toISOString();`

or set by a redux action in

`dispatch(fetchDatesSuccess(dates, moment().toISOString()));`

Also, from redux dev tools I can see it's persist/REHYDRATE that sets it to null

![](https://imgur.com/a/w3n97)

This is the first action that redux dev tools inspector shows after purging from another tab.
